### ISSUE TITLE
sam sync goes into "running incremental build" loop

### ISSUE THREAD

**Issue by maplion** [aws/aws-sam-cli#3443]
### Description:
After I successfully sync with sam sync, as soon as I do anything in VSCode (even before saving any changes), it will begin going into this "running incremental build" loop and continue to cycle:

![image](https://user-images.githubusercontent.com/8908849/140407806-9f603566-8fab-4354-b2fb-58016d2fcc24.png)

### Steps to reproduce:
Run `sam sync --stack-name <stack name> --watch --profile <profile> --region us-east-1`

### Observed result:
Answer "y" when asked if you want to proceed, then watch Cloudformation initiate deployment, then says "Infra sync completed".  After that, it sits idle, as it should, but as soon as I click into the template.yml, it starts looping.

### Expected result:
It should remain idle until I save something.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 10
2. `sam --version`: 1.35.0
3. AWS region: us-east-1

`Add --debug flag to command you are running`

Did this and it is constantly spamming the following (note StageName and LayerArn are Parameters):

2021-11-04 13:41:29,579 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,599 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,599 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,618 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,637 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,637 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,656 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,674 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,674 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,693 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,712 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,712 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,731 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,749 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,749 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,767 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,788 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,789 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,865 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,883 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,884 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,906 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,925 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,925 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,943 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,968 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:29,968 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:29,987 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,004 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,005 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,025 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,044 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,044 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,065 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,084 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,085 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,105 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,124 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,125 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,145 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,166 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,166 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,186 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,205 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,205 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,223 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,245 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,245 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,263 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,283 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,283 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,303 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,322 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,322 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,343 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,361 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,361 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,379 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,397 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,397 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,418 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,437 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,438 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,457 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,478 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,479 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,497 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,515 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,516 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,538 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,558 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,558 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,577 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,596 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,596 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,615 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,637 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,637 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,655 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,674 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,675 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,693 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,714 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,715 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,737 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,754 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,755 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,773 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,791 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,792 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,814 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,843 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,844 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,863 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,885 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,885 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,908 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,926 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,927 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,946 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,964 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:30,965 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:30,983 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,001 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,001 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,019 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,038 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,039 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,057 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,076 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,077 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,095 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,114 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,114 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,135 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,155 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,155 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,174 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,192 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,193 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,213 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,233 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,234 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,253 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,271 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,272 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,290 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,311 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,311 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,331 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,352 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,353 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,371 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,390 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,390 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,409 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,428 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,428 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,447 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,465 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,466 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,485 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,502 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,503 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,522 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,542 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,543 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,561 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,580 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,580 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,599 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,618 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,618 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,639 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,657 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,658 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,679 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,700 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,701 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,720 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,751 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,752 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,775 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,795 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,796 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,821 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,840 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,841 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,861 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,878 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,878 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,897 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,916 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,917 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,943 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,965 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:31,966 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:31,985 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,003 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,003 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,022 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,041 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,041 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,061 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,082 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,083 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,104 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,126 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,127 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,152 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,172 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,173 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,193 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,212 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,212 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,231 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,248 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,249 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,267 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}
2021-11-04 13:41:32,290 | Found the same SyncFlow in queue. Skip adding.
2021-11-04 13:41:32,290 | Collected default values for parameters: {'StageName': 'v1', 'LayerArn': 'v1'}

**Comment by CoshUS** [comment]
Hey @maplion,
Is your Function CodeURI the project root directory?

**Comment by maplion** [comment]
@CoshUS Yes, that is correct.

**Comment by CoshUS** [comment]
Got it, thanks. The root cause is due to .aws-sam folder being watched as function code. When the function builds, artifacts will be generated into the .aws-sam/build folder which triggers another sync. We are planning to fix this by excluding .aws-sam folder.
In the meantime, the workaround is to use a specific folder for function code instead of project root directory.

**Comment by maplion** [comment]
Great, thank you.

**Comment by maplion** [comment]
@CoshUS Your suggested workaround seems to work for me.

**Comment by maplion** [comment]
@CoshUS How can I make this work with Python?  Putting things in a subfolder complicates Python imports and the AWS Knowledge center suggests having them in the root.

**Comment by moelasmar** [comment]
@maplion .. I do not think you need to have multiple subfolders .. you just need to copy all python code and put it in in one folder, and then refer to it in the Lambda function.

Can you share your project structure so we can understand more your problem, and also please share with us the Knowledge center suggests you are referring to.

**Comment by maplion** [comment]
I was able to get this to work by having my lambda_handler.py in the root
of my project and other python in packages (i.e. folders) as suggested.

**Comment by moelasmar** [comment]
Thanks @maplion for your reply, so does it mean that your issue got resolved?

**Comment by maplion** [comment]
@moelasmar I stand corrected; I still seem to be getting this behavior on occasion.  Here is my project structure:

.
├── authorizer
│   ├── __init__.py
│   └── global_admin.py
├── buildspec.yml
├── handlers
│   ├── __init__.py
│   ├── constants.py
│   ├── db_handler.py
├── lambda_handler.py
├── package.json
├── requirements-dev.txt
├── requirements.txt
└── template.yml

**Comment by moelasmar** [comment]
@maplion .. could  you please share the Lambda Function definition in your template file.

**Comment by maplion** [comment]
```yaml
Resources:
  OcLambdaAuthorizerFunction:
    Type: 'AWS::Serverless::Function'
    Properties:
      Handler: lambda_handler.handler
      FunctionName: lambda-authorizer
      Role: '<my role>'
      Events:
        ExplicitApi:
          Type: HttpApi
          Properties:
            Path: /{proxy+}
            Method: ANY
            TimeoutInMillis: 15000
```

**Comment by moelasmar** [comment]
You can create a sub directory for your lambda function, let's say `src`, and move your code to `src` directory.
So your project structure can be like the following
```
.
├── src
│   └── authorizer
│   │   ├── __init__.py
│   │   └── global_admin.py
│   └── handlers
│   │   ├── __init__.py
│   │   ├── constants.py
│   │   └── db_handler.py
│   └── lambda_handler.py
│   └── requirements-dev.txt
│   └── requirements.txt
├── buildspec.yml
├── package.json
└── template.yml
```

and change your Lambda function definition in your template file like the following:
```
Resources:
  OcLambdaAuthorizerFunction:
    Type: 'AWS::Serverless::Function'
    Properties:
      CodeUri: src/
      Handler: lambda_handler.handler
      FunctionName: lambda-authorizer
      Role: '<my role>'
      Events:
        ExplicitApi:
          Type: HttpApi
          Properties:
            Path: /{proxy+}
            Method: ANY
            TimeoutInMillis: 15000
```
Could you please try my suggestion and let me know if it solved your issue or not.

**Comment by maplion** [comment]
@moelasmar I'm not sure how this is to work if my CodeUri is a zip file for my deployment (I can't just tag /src/ at the end and make it work, I don't think).

**Comment by moelasmar** [comment]
if your code uri is a zip file, you can normally do it like the following:
```
Resources:
  OcLambdaAuthorizerFunction:
    Type: 'AWS::Serverless::Function'
    Properties:
      CodeUri: src/func_src.zip
      Handler: lambda_handler.handler
      FunctionName: lambda-authorizer
      Role: '<my role>'
      Events:
        ExplicitApi:
          Type: HttpApi
          Properties:
            Path: /{proxy+}
            Method: ANY
            TimeoutInMillis: 15000
```
and for deployment, you can use `sam deploy` and it will take care of everything, it will package the lambda function source code, zip it, and upload it to S3.

**Comment by maplion** [comment]
@moelasmar That won't work for me because I am not using `sam deploy`; I am self-packaging by downloading dependencies, unzipping, and then creating my own zip and sending it up to Octopus Deploy as a full package (rather than just a template.yml file), and then referencing the template in the deploy like this when deploying:
```
CodeUri: "s3://<bucket name>/my-package@S1.0.0.29@683DC7D6EA0532488E5736104F6D9771.zip"
```

**Comment by CoshUS** [comment]
Released in SAM CLI 1.36.0
